flocx_market/matcher.py

```diff
@@ -0,0 +1,107 @@
+import jmespath
+import re
+
+
+def apply_operator(val1, val2, op):
+
+    if val1 is None or val2 is None or op is None:
+        return False
+
+    # null operator
+    if op == 'null':
+        return val1
+
+    all_num_ops = ['==', '!=', '<', '<=', '>', '>=']
+    all_str_ops = ['eq', 'ne', 'startswith', 'endswith', 'matches']
+    all_list_ops = ['in', 'contains']
+
+    # all numeric operations
+    if op in all_num_ops:
+
+        try:
+            val1 = float(val1)
+            val2 = float(val2)
+
+        except ValueError as verr:
+            print (verr.args)
+            return False
+
+        if op == '==':
+            return val1 == val2
+
+        if op == '!=':
+            return val1 != val2
+
+        if op == '>':
+            return val1 > val2
+
+        if op == '>=':
+            return val1 >= val2
+
+        if op == '<':
+            return val1 < val2
+
+        if op == '<=':
+            return val1 <= val2
+
+    # string operations
+    if op in all_str_ops:
+        try:
+            val1 = str(val1)
+            val2 = str(val2)
+        except ValueError as verr:
+            print (verr.args)
+            return False 
+        if op == 'eq':
+            return val1 == val2
+
+        if op == 'ne':
+            return val1 != val2
+
+        if op == 'startswith':
+            return val1.startswith(val2)
+
+        if op == 'endswith':
+            return val1.endswith(val2)
+
+        if op == 'matches':
+            if re.search(val2, val1):
+                return True
+            else:
+                return False
+
+    # list operators
+    if op in all_list_ops:
+
+        if op == 'in':
+            try:
+                val1 = str(val1)
```

Comment:
I don't think we need explicit casting here; someone may conceivably be checking if a number is in a list, and this casting would make that test fail.